app/views/application/_navbar.haml

```diff
@@ -10,14 +10,14 @@
             New event
         %li
           %p.navbar-text
+            = image_tag current_user.image, size: '20x20', alt: "Profile image"
             = current_user.name
-            
-            = link_to logout_path do
-              %i.fa.fa-sign-out
-              Sign out
+        %li
+          = link_to logout_path do
+            %i.fa.fa-sign-out
```

Comment:
In gem `font-awesome-sass` there is a helper method `icon 'plus', 'Add something', class: 'my-class', id: 'my-id'` which will output icon with text. You could omit the `'Add something'` string to output just icon in your view.